WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/ReaderViewPage.java

```diff
@@ -0,0 +1,106 @@
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiSelector;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.wordpress.android.support.WPSupportUtils.DEFAULT_TIMEOUT;
+import static org.wordpress.android.support.WPSupportUtils.isTextDisplayed;
+import static org.wordpress.android.support.WPSupportUtils.swipeToLeft;
+import static org.wordpress.android.support.WPSupportUtils.swipeToRight;
+
+public class ReaderViewPage {
+    UiDevice mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+    String mLikerContainerId = "org.wordpress.android.prealpha:id/liker_faces_container";
+    String mRelatedPostsId = "org.wordpress.android.prealpha:id/container_related_posts";
+
+    UiObject mLikerContainer = mDevice.findObject(new UiSelector().resourceId(mLikerContainerId));
+    UiObject mRelatedPostsContainer = mDevice.findObject(new UiSelector().resourceId(mRelatedPostsId));
+    UiObject mSwipeForMore = mDevice.findObject(new UiSelector().textContains("Swipe for more"));
+
+    public ReaderViewPage waitUntilLoaded() {
+        mRelatedPostsContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage like() {
+        tapLikeButton();
+        mLikerContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage unlike() {
+        tapLikeButton();
+        mLikerContainer.waitUntilGone(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    private void tapLikeButton() {
+        mSwipeForMore.waitUntilGone(DEFAULT_TIMEOUT);
+        // Even though it was working locally in simulator, tapping the footer buttons,
+        // like 'mLikeButton.click()', was not working in CI.
+        // The current workaround is to use arrows navigation.
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_DOWN);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_CENTER);
+        resetFocus();
+    }
+
+    private void resetFocus() {
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+    }
+
+    public ReaderPage goBackToReader() {
+        mDevice.pressBack();
+
+        return new ReaderPage();
+    }
+
+    public ReaderViewPage slideToPreviousPost() {
+        swipeToRight();
+
+        return this;
+    }
+
+    public ReaderViewPage slideToNextPost() {
+        swipeToLeft();
+
+        return this;
+    }
+
+    public ReaderViewPage verifyPostDisplayed(String title) {
+        assertTrue("Post title was not displayed. Target post: " + title, isTextDisplayed(title));
+
+        return this;
+    }
```

Comment:
I was also checking for the post content here, however it was making the test super flaky (in CI), possibly because it was in the WebView. Many times the test would fail saying the content was not displayed while it was clearly visible from the Firebase video.
Even though the title is the only element in the assertion as of now, while opening a post we already wait for the related posts to be loaded in `ReaderViewPage().waitUntilLoaded()`.